File: android/src/main/java/com/getcapacitor/community/classes/options/EstablishOptions.java
package com.getcapacitor.community.classes.options;

import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;

public class EstablishOptions {

    @Nullable
    private String peerID;

    public EstablishOptions(PluginCall call) {
        @Nullable
        String peerID = call.getString("peerID");
        this.setPeerID(peerID);
    }

    private void setPeerID(@Nullable String peerID) {
        this.peerID = peerID;
    }

    @Nullable
    public String getPeerID() {
        return peerID;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Options/EstablishOptions.swift
import Foundation
import Capacitor

@objc public class EstablishOptions: NSObject {
    private var peerID: String?

    init(_ call: CAPPluginCall) {
        super.init()

        if let peerID = call.getString("peerID") {
            self.setPeerID(peerID)
        }
    }

    func setPeerID(_ peerID: String?) {
        self.peerID = peerID
    }

    func getPeerID() -> String? {
        return self.peerID
    }
}
